contributors/devel/sig-architecture/feature-gates.md

```diff
@@ -1,87 +1,256 @@
+with some grace period for the transition.
 
-[After at least two releases post-GA and deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation), the feature gate is removed. Typically, we add a comment in [kubefeatures.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go) such as: `// remove in 1.23` to signal when we plan to remove the feature gate. Remember when the feature gate is removed and the deployer has forgotten to drop the reference to the feature in the CLI flags (say the `kube-apiserver`), then they will see a hard failure.
+[After at least two releases post-GA and deprecation](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation),
+feature gates should be removed. Typically, we add a comment in
+[the code](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go)
+such as: `// remove in 1.23` to signal when we plan to remove the feature gate.
+We provide this grace period to give users time to stop referencing "finished"
+gates. If a feature gate is removed and a user has forgotten to drop the
+reference to it (e.g. in the CLI flags of `kube-apiserver`), then they will see
+a hard failure.
 
-Also note that when we set `LockToDefault` to `true`, we remove all references (if/then conditions) to the feature gate from the codebase.
+When we set `LockToDefault` to `true`, we also remove all references to the feature
+gate from the codebase.
 
-## Deprecation
+### Deprecation
 
 * `PreRelease` is set to `featuregate.Deprecated`
 * `Default` is set to `false`
 * See [Kubernetes Deprecation Policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation) for more details
 
-Very rarely we will deprecate some aspect of Kubernetes (almost always something that has no actual impact).  When we do that, the pattern is to name the gate so it describes the functionality being deprecated and to default the value to `false`.  If some user is impacted by the deprecation, they can set that gate to `true` to unbreak themselves (and then file a bug).  If this happens, we must reconsider the deprecation and may choose to abandon it entirely by changing the gate back to `true` for a release or two and eventually removing it.
+Very rarely we will deprecate some user-visible aspect of Kubernetes (almost
+always something that has no actual impact or which is truly a bug). When we
+do that, we also use a feature gate. The pattern is to name the gate so it
+describes the functionality being deprecated and to default the value to
+`false`.
+
+If some user is impacted by the deprecation, they can set that gate to `true`
+to unbreak themselves (and then file a bug). If this happens, we must
+reconsider the deprecation and may choose to abandon it entirely by changing
+the gate back to `true` for a release or two and eventually removing it.
+
+NOTE: We do not generally remove fields from the API.
```

Comment:
removing fields that are GA on a GA API is a breaking change, isn't it? 
I think we can just not do it, we discussed this with some fields like service.spec.LoadBalancerIP

Reply:
yes, we just don't do it.